src/workloads/query/QueryStats.yml

```diff
@@ -10,233 +10,176 @@ KeyWords:
+    Parameters:
+      MaxPhase: *MaxPhase
+      Phases: [
+        13, 23, 33, 43,
+        16, 26, 36, 46,
+      ]
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: PrepopulateActor  # Prepopulate the query stats cache until it's full. This is to simulate
+                           # the steady-state scenario, which is what we're mostly interested in.
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phases: [14, 24, 34, 44]
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: RateLimitActor  # Reset the rate limit to the desired value. This needs to be kept
+                         # up-to-date with the actual rate limit.
+    Parameters:
+      MaxPhase: *MaxPhase
+      Phases: [15, 25, 35, 45]
+      RateLimitHz: *RateLimitHz
+
+# No concurrent $queryStats query, all find queries are roughly the same size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 17
+      Payload: *FixedPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Base-Fixed"
+
+# Concurrent HMAC-transformed $queryStats query, all find queries are roughly the same size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 27
+      Payload: *FixedPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Fixed"
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: QueryStatsActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 27
+      QueryStatsParameters: *HmacParameters
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Fixed"
+
+# No concurrent $queryStats query, find queries vary widely in size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 37
+      Payload: *RandomPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Base-Random"
+
+# Concurrent HMAC-transformed $queryStats query, find queries vary widely in size.
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: FindCommandActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 47
+      Payload: *RandomPayload
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Random"
+- LoadConfig:
+    Path: ../../phases/query/QueryStats.tmpl
+    Key: QueryStatsActor
+    Parameters:
+      CacheSizeMb: *CacheSizeMb
+      MaxPhase: *MaxPhase
+      Phase: 47
+      QueryStatsParameters: *HmacParameters
+      RateLimitHz: *RateLimitHz
+      Suffix: "Hmac-Random"
 
 AutoRun:
 - When:
     mongodb_setup:
       $eq:
       - standalone-all-feature-flags
       - standalone-query-stats
```

Comment:
I think that the first 3 in this list will all provide the same data, yeah? I think we can get away with just running one of these 3, and doing the same for the other platforms. I think maybe just do the non-query-stats ones? That seems like the end state we want, so we won't have to change it up later.

Reply:
Ideally, they'd provide the same data, but this isn't a given. The rate-limited query stats variants run with a different server parameter. I'm of the opinion that server parameter differences should still be tested to make sure that the performance doesn't change. This adds an extra layer of protection and information in case something _does_ create a performance difference due to the rate limit parameter.